acceptance/tests/windows/enable_password_changes_special_users.rb

```diff
@@ -0,0 +1,91 @@
+test_name 'Puppet should change passwords for disabled, expired, or locked out Windows user accounts' do
+  require 'date'
+  require 'puppet/acceptance/windows_utils'
+
+  extend Puppet::Acceptance::WindowsUtils
+  confine :to, platform: 'windows'
+
+  def random_username
+    "pl#{rand(999999).to_i}"
+  end
+
+  def change_password_manifest(username)
+    return <<-MANIFEST
+    user { '#{username}':
+      ensure   => 'present',
+      password => '#{NEW_PASSWORD}' 
+    }
+    MANIFEST
+  end
+
+  INITIAL_PASSWORD='1nitialP@ssword'
+  NEW_PASSWORD="Password-#{rand(999999).to_i}"
+
+  agents.each do |host|
+    disabled_username = random_username
+
+    step "Create a disabled user account" do
+      on(host, "cmd.exe /c net user #{disabled_username} #{INITIAL_PASSWORD} /active:no /add")
+    end
+
+    step "Change the disabled user account's password with puppet" do
+      apply_manifest_on(host, change_password_manifest(disabled_username))
+    end
+
+    step "Enabling the user account as the AccountManagement context can't verify disabled users" do
```

Comment:
The [PS script](https://github.com/puppetlabs/puppet/blob/5.5.x/acceptance/lib/puppet/acceptance/windows_utils.rb#L20) that checks the user credentials only work if the user is not disabled, expired or locked out. 
Activating the user before the script ensures that puppet changed the password.